File: component-service/src/main/java/dev/yidafu/pan/component/convertor/ComponentAttributeConvertor.java
package dev.yidafu.pan.component.convertor;

import com.fasterxml.jackson.databind.util.JSONPObject;
import dev.yidafu.pan.component.domain.dto.SaveComponentAttributeDTO;
import dev.yidafu.pan.component.domain.entity.ComponentAttribute;
import org.mapstruct.Mapper;

import java.util.List;



@Mapper(componentModel = "spring")
public interface ComponentAttributeConvertor {

    List<ComponentAttribute> from(List<SaveComponentAttributeDTO> list);

    ComponentAttribute from(SaveComponentAttributeDTO list);
}

File: component-service/src/main/kotlin/dev/yidafu/pan/component/convertor/ComponentAttributeConvertor.kt
package dev.yidafu.pan.component.convertor

import dev.yidafu.pan.common.json.JsonValue
import dev.yidafu.pan.common.model.dto.SaveComponentAttributeDTO
import dev.yidafu.pan.common.model.dto.UpdateComponentAttributeDTO
import dev.yidafu.pan.component.domain.model.ComponentAttribute
import org.mapstruct.InjectionStrategy
import org.mapstruct.Mapper

@Mapper(componentModel = "spring", injectionStrategy = InjectionStrategy.CONSTRUCTOR)
interface ComponentAttributeConvertor {

    fun from(list: List<SaveComponentAttributeDTO>): List<ComponentAttribute>

    fun from(list: SaveComponentAttributeDTO): ComponentAttribute

    companion object {
        fun to(ownerId: Long, map: Map<String, JsonValue>): List<SaveComponentAttributeDTO> {
            return map.map {
                SaveComponentAttributeDTO(
                    ownerId,
                    it.key,
                    it.value.valueType?.value,
                    it.value.value
                )
            }.toList()
        }
        //
        fun toUpdateDTOList(ownerId: Long, map: Map<String, JsonValue>): List<UpdateComponentAttributeDTO> {
            return map.map {
                UpdateComponentAttributeDTO(
                    ownerId,
                    it.key,
                    it.value.valueType?.value,
                    it.value.value
                )
            }.toList()
        }
    }
}
